src/cascadia/TerminalControl/XamlUiaTextRange.cpp

```diff
@@ -0,0 +1,176 @@
+            THROW_IF_FAILED(SafeArrayAccessData(pReturnVal, (void**)&pVals));
+
+            long lBound, uBound;
+            THROW_IF_FAILED(SafeArrayGetLBound(pReturnVal, 1, &lBound));
+            THROW_IF_FAILED(SafeArrayGetUBound(pReturnVal, 1, &uBound));
+
+            long count = uBound - lBound + 1;
+
+            std::vector<double> vec;
+            vec.reserve(count);
+            for (int i = 0; i < count; i++)
+            {
+                double element = pVals[i];
+                vec.push_back(element);
+            }
+
+            winrt::com_array<double> result{ vec };
+            returnValue = std::move(result);
+        }
+        catch (...)
+        {
+        }
+    }
+
+    Windows::UI::Xaml::Automation::Provider::IRawElementProviderSimple XamlUiaTextRange::GetEnclosingElement()
+    {
+        return _parentProvider;
+    }
+
+    winrt::hstring XamlUiaTextRange::GetText(int32_t maxLength) const
+    {
+        BSTR returnVal;
+        _uiaProvider->GetText(maxLength, &returnVal);
+        return winrt::to_hstring(returnVal);
+    }
+
+    int32_t XamlUiaTextRange::Move(Windows::UI::Xaml::Automation::Text::TextUnit unit,
+                                   int32_t count)
+    {
+        int returnVal;
+        _uiaProvider->Move(static_cast<::TextUnit>(unit),
+                           count,
+                           &returnVal);
+        return returnVal;
+    }
+
+    int32_t XamlUiaTextRange::MoveEndpointByUnit(Windows::UI::Xaml::Automation::Text::TextPatternRangeEndpoint endpoint,
+                                                 Windows::UI::Xaml::Automation::Text::TextUnit unit,
+                                                 int32_t count) const
+    {
+        int returnVal;
+        _uiaProvider->MoveEndpointByUnit(static_cast<::TextPatternRangeEndpoint>(endpoint),
+                                         static_cast<::TextUnit>(unit),
+                                         count,
+                                         &returnVal);
+        return returnVal;
+    }
+
+    void XamlUiaTextRange::MoveEndpointByRange(Windows::UI::Xaml::Automation::Text::TextPatternRangeEndpoint endpoint,
+                                               Windows::UI::Xaml::Automation::Provider::ITextRangeProvider pTargetRange,
+                                               Windows::UI::Xaml::Automation::Text::TextPatternRangeEndpoint targetEndpoint) const
+    {
+        auto self = winrt::get_self<XamlUiaTextRange>(pTargetRange);
+        _uiaProvider->MoveEndpointByRange(static_cast<::TextPatternRangeEndpoint>(endpoint),
+                                          /*pTargetRange*/ self->_uiaProvider.get(),
+                                          static_cast<::TextPatternRangeEndpoint>(targetEndpoint));
+    }
+
+    void XamlUiaTextRange::Select() const
+    {
+        THROW_IF_FAILED(_uiaProvider->Select());
+    }
+
+    void XamlUiaTextRange::AddToSelection() const
+    {
+        // we don't support this

```

Comment:
But will we eventually? It sounds reasonable. Why don't we?

Reply:
From msdn: _Adds to the collection of highlighted text in a text container that supports multiple disjoint selections._

Personally, I don't think we'll ever do that, But either way, we don't have this in UiaTextRange today and this is just a wrapper.